src/reconcile/rec_row.c

```diff
@@ -868,7 +868,8 @@ __wt_rec_row_leaf(
                 if (F_ISSET(S2C(session), WT_CONN_HS_OPEN) && !WT_IS_HS(btree)) {
                     WT_ERR(__wt_row_leaf_key(session, page, rip, tmpkey, true));
                     /* Start from WT_TS_NONE to delete all the history store content of the key. */
-                    WT_ERR(__wt_hs_delete_key_from_ts(session, btree->id, tmpkey, WT_TS_NONE));
+                    WT_ERR(__wt_hs_delete_key_from_ts(
+                      session, tw.stop_txn, btree->id, tmpkey, WT_TS_NONE));
```

Comment:
The number of different ways we get the new transaction ID being passed into this function worries me. It comes from `tw` here, `upd`, `first_non_ts_upd`, and `oldest_upd`. It's not really clear to me what it means, or how we could know that we are passing the right thing in.
Is there some other way we could make this change to get more consistency? (no is a reasonable answer, but I thought it was worth asking the question).

Reply:
I don't think we can avoid that in this instance without a decent amount of refactoring which is being discussed here: https://github.com/wiredtiger/wiredtiger/pull/5807#discussion_r439942118. 

I personally would rather go with this change as is, barring a few extra PR fixes and then refactor, OR make the visibility and double tombstone change, remove the txnid change and then do that with the refactor as a separate ticket to avoid this ticket becoming too much of a change.

Reply:
In short, we should pass in the txnid of the newest mixed mode operation on the update chain. Currently, it comes from different sources.

Reply:
I'll move my question up to that thread.